source/tests/system/nixnet_utilities.h

```diff
@@ -0,0 +1,195 @@
+#include <gtest/gtest.h>
+#undef interface
+#include <nixnet/nixnet_client.h>
+
+#include <string>
+#include <vector>
+
+#define EXPECT_SUCCESS(response_)    \
+  ([](auto& response) {              \
+    EXPECT_EQ(0, response.status()); \
+    return response;                 \
+  })(response_)
+
+#define EXPECT_XNET_ERROR(error, response) \
+  if (1) {                                 \
+    EXPECT_EQ(error, (response).status()); \
+  }
+
+namespace nixnet_utilities {
+namespace {
```

Comment:
nit: I don't think anonymous namespaces work in headers because they get pulled into the including translation unit. In this case, Normally they would be used in cpp files to avoid defining stuff in shared/global namespaces and potentially violating the One Definition Rule.

Here are the relevant core guidelines:
* https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#Rs-unnamed
* https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#Rs-unnamed2

You can use a `detail` namespace for a similar purpose, that's what I'd use for a shared header-only library. I also use the `_` prefix or postfix in headers that are only used internally (like you're doing below).

In this case, I don't see where the methods are called, so I'm not sure what the intent is.

Reply:
Speaking of the ODR, you also probably need to inline these methods. At least before they get included twice.